pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -330,6 +160,39 @@ func (g *Gatherer) GetGatheringFunctions(ctx context.Context) (map[string]gather
 	return gatheringFunctions, nil
 }
 
+// fetchGatheringRulesFromServer returns the latest version of the rules from the server
+func (g *Gatherer) fetchGatheringRulesFromServer(ctx context.Context) ([]GatheringRule, error) {
+	gatheringRulesJSON, err := g.getGatheringRulesJSON(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return parseGatheringRules(gatheringRulesJSON)
+}
+
+// getGatheringRulesJSON returns json version of the rules from the server
+func (g *Gatherer) getGatheringRulesJSON(ctx context.Context) (string, error) {
+	if g.configurator == nil {
+		return "", fmt.Errorf("no configurator was provided")
+	}
+
+	config := g.configurator.Config()
+	if config == nil {
+		return "", fmt.Errorf("config is nil")
+	}
+
+	if g.gatheringRulesServiceClient == nil {
+		return "", fmt.Errorf("gathering rules service client is nil")
+	}
+
+	rulesBytes, err := g.gatheringRulesServiceClient.RecvGatheringRules(ctx, config.ConditionalGathererEndpoint)
```

Comment:
What about adding some info log around here please? Something like fetching or pulling the conditions from endpoint bla bla bla?

Reply:
done

Reply:
I thought it would be good to put the endpoint URL in the log. Theoretically, it can happen that someone sets wrong endpoint and then it can help when troubleshooting.

Reply:
done